src/Shared/ErrorUtilities.cs

```diff
@@ -467,7 +468,7 @@ internal static void VerifyThrowArgument([DoesNotReturnIf(false)] bool condition
         /// Throws an argument out of range exception.
         /// </summary>
         [DoesNotReturn]
-        internal static void ThrowArgumentOutOfRange(string parameterName)
+        internal static void ThrowArgumentOutOfRange(string? parameterName)
```

Comment:
Why is this needed? - I'd expect this change to not cause weaker nullability guarantee

Reply:
Yeah this is a bit unfortunate. To use `[CallerMemberName]` you have to provide a default value, and the one for `string parameterName` that makes the most sense IMO is `null`. But then there's a possibility that `null` flows through the other overload of `ThrowArgumentOutOfRange` to here.

I think I prefer this to swapping it for `""` but if you feel strongly the other way I can swap it. Definitely not a strong preference on my side.

Reply:
Oh - makes sense.
I'm fine with this.